lib/veri_report.mli

```diff
@@ -0,0 +1,34 @@
+open Core_kernel.Std
+
+(**  Report module.
+
+     Report is intended to get some quantitive results: 
+     - number of successful comparisons;
+     - number of unsuccessful comparisons;
+     - number of instructions, that weren't recognized by disasm;
+     - number of instructions, that weren't lifted by BIL;
+     - histogram  as mapping from instruction name to number of 
+       unsuccessful comparison casses.
+*)
+
+type hist = (string * int) list [@@deriving bin_io, sexp]
+
+type succ = [ 
+  | `Right 
+  | `Undsm 
+  | `Unlif of string
+  | `Wrong of string 
+]
+
+type t [@@deriving bin_io, sexp]
+
+val create: unit -> t
+val right: t -> int
```

Comment:
if you have a value for result, why do you need to provide a function for each possible branch of your result type? You should use `val count : succ -> int`.

But in any case, this `succ` type is wrong, and the whole module doesn't make many sense.